### PR TITLE
Recipe app: the final PR 

### PR DESCRIPTION
**Pull request by naji-97**
- [ ] A navigation menu that allows users to open all of the pages I created.

- [ ] A login page using Devise.

- [ ] A registration page using Devise.

- [ ] A food list (with all CRUD implementation, except for 'update'), displays a list and quantity of a given food of food added by the logged-in user.

- [ ] A recipes list (with all CRUD implementation, except for 'update'),  display a list of recipes created by the logged-in user, each recipe leads to recipe details, and the user can delete the recipe if it's the owner.

- [ ] A general shopping list shows the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user) and counts the total food items and total price of the missing food.

- [ ] A list of all public recipes from other users with their names and total prices display a list of all public recipes ordered by the newest lead to recipe details.

- [ ] Recipe details: displays a toggle button that allows for a recipe to be made public or private.
        If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
        If the user is the owner of the recipe, should lead to the form that allows the user to add new food.

- [ ]  Write unit and integration tests using RSpec and Capybara testing libraries.

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    @recipe = Recipe.new
+  end
+
+  def public_recipes
+    @recipes = Recipe.where(public: true).order(created_at: :desc)
+  end
+
+  def show
+    @recipe = Recipe.includes(:user, recipe_foods: :food).find(params[:id])
+  end
```
Comment (OyePriscilla) [inline app/controllers/recipes_controller.rb:18]: - [ ] I observed that you ommitted two things as listed in the wireframe, `generate_shopping_list `and `add_ingrdient`. Kindly implemented as expected. 👍

 ## Wireframe
Untitled

## Your design
Untitled

File: app/controllers/foods_controller.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  end
+
+  def destroy
+    @food = current_user.foods.find(params[:id])
+    if @food.destroy
+      redirect_to foods_path, notice: 'Food item deleted successfully.'
+    else
+      redirect_to foods_path, alert: 'Failed to delete this item'
+    end
+  end
```
Comment (OyePriscilla) [inline app/controllers/foods_controller.rb:33]: - [ ] I was unable to delete recipe, I got the error screen shot below 👇, it indicates that there's an issue with a `has_many :through` association in your ActiveRecord models. Kindly fix it. 👍 🔴 **Please take note that I got the same error when I tried to delete a food item** 🔴

## Error from Recipe_show page
Untitled

## Error from Food page
Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @naji-97 and @Yacoubou-seidou 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Set up Devise for authentication ✔️
Set up associations between models ✔️ 
Used params from browser requests in a safe way ✔️ 
Used preprocessed HTML file with embedded Ruby code ✔️ 
Used layouts and templates for shared content ✔️ 
Used database migration files to maintain database schema ✔️ 
Used validations for models ✔️ 
Secured app from n+1 problems ✔️ 
Implemented tests for the app ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi Team, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


### Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**